--- gallery-plus.php_oud	2010-10-06 10:38:12.182987001 +0200
+++ gallery-plus.php	2010-10-06 11:34:26.295373002 +0200
@@ -1,9 +1,9 @@
 <?php
 /*
-Plugin Name: Gallery Plus Oud
+Plugin Name: Gallery Plus
 Plugin URI: http://www.hawkwood.com/archives/category/wordpress/plugins/gallery-plus
-Description: This plugin overrides the built-in Wordpress gallery to enable additional options. 
-Author: Justin Hawkwood
+Description: AANGEPAST This plugin overrides the built-in Wordpress gallery to enable additional options. 
+Author: Justin Hawkwood, aangepast door Ben Bridts
 Author URI: http://www.hawkwood.com/
 Version: 1.4.1
 */
@@ -113,6 +113,7 @@
 		'captiontag' => 'dd',
 		'columns'    => get_option('gallery_plus_columns'),
 		'size'       => get_option('gallery_plus_size'),
+		'fullsize'   => get_option('gallery_plus_fullsize'),
 //		'link2full'  => get_option('gallery_plus_link2full'),
 		'linktype'  => get_option('gallery_plus_linktype'),
 		'titlecaption'  => get_option('gallery_plus_titlecaption'),
@@ -140,7 +141,11 @@
 		foreach ($adhoc as $id) {
 			$attachments[$id] = get_post($id);
 		}
-	} else $attachments = get_children("post_parent=$id&post_type=attachment&post_mime_type=image&orderby={$orderby}");
+	} else {
+		$attachments = get_children("post_parent=$id&post_type=attachment&post_mime_type=image&orderby={$orderby}");
+		//fix om de afbeeldingen in omgekeerde volgorde te tonen, Mante
+		$attachments = array_reverse($attachments);
+	}
 //	$attachments = get_children( array('post_parent' => $id, 'post_status' => 'inherit', 'post_type' => 'attachment', 'post_mime_type' => 'image', 'order' => $order, 'orderby' => $orderby) );	// WP 2.6 uses this
 
 	if ( empty($attachments) )
@@ -182,7 +187,8 @@
 		} else if ($linktype == 'image') {
 			// show full res link to image
 			$thumbnail_link = wp_get_attachment_image_src($attachment->ID, $size, false);
-			$full_image_href = wp_get_attachment_image_src($attachment->ID, 'full', false);
+			// Gebruik fullsize-optie ipv 'full' in de links.
+			$full_image_href = wp_get_attachment_image_src($attachment->ID, $fullsize, false);
 //			$img_title = trim(htmlspecialchars($attachment->post_title, ENT_QUOTES ));
 //			$img_caption = trim(htmlspecialchars($attachment->post_excerpt, ENT_QUOTES));
 			if ($titlecaption == 'none') {
@@ -197,6 +203,11 @@
 			} else if ($titlecaption == 'caption') {
 				$img_title = trim(htmlspecialchars($attachment->post_excerpt, ENT_QUOTES));
 				$img_caption = trim(htmlspecialchars($attachment->post_excerpt, ENT_QUOTES));
+			} else if ($titlecaption == 'download') {
+			    $download_link = wp_get_attachment_image_src($attachment->ID, 'full', false);
+				$img_title = trim(htmlspecialchars($attachment->post_title, ENT_QUOTES));
+				$img_title .= " <a href='{$download_link[0]}'>download origineel</a>";
+				$img_caption = ''; //trim(htmlspecialchars($attachment->post_excerpt, ENT_QUOTES));
 			} else {
 				$img_title = trim(htmlspecialchars($attachment->post_title, ENT_QUOTES ));
 				$img_caption = trim(htmlspecialchars($attachment->post_excerpt, ENT_QUOTES));
