--- options.php_oud	2010-10-06 10:58:58.092987002 +0200
+++ options.php	2010-10-06 11:34:38.415373002 +0200
@@ -5,6 +5,7 @@
 if ('process' == $_POST['stage']) {
     update_option('gallery_plus_columns', trim($_POST['gallery_plus_columns']));
     update_option('gallery_plus_size', $_POST['gallery_plus_size']);
+    update_option('gallery_plus_fullsize', $_POST['gallery_plus_fullsize']);
 //	update_option('gallery_plus_link2full', $_POST['gallery_plus_link2full']);
 	update_option('gallery_plus_linktype', $_POST['gallery_plus_linktype']);
 	update_option('gallery_plus_titlecaption', $_POST['gallery_plus_titlecaption']);
@@ -21,6 +22,7 @@
 /*Get options for form fields */
 $gallery_plus_columns = stripslashes(get_option('gallery_plus_columns'));
 $gallery_plus_size = stripslashes(get_option('gallery_plus_size'));
+$gallery_plus_fullsize = stripslashes(get_option('gallery_plus_fullsize'));
 /*
 if (get_option('gallery_plus_link2full') !== false) delete_option('gallery_plus_link2full');
 $gallery_plus_link2full = stripslashes(get_option('gallery_plus_link2full'));
@@ -66,6 +68,7 @@
         <td><select name="gallery_plus_size">
 			<option value="thumbnail"<?php if ($gallery_plus_size == 'thumbnail') echo ' selected="selected"'; ?>>Thumbnail</option>
 			<option value="medium"<?php if ($gallery_plus_size == 'medium') echo ' selected="selected"'; ?>>Medium</option>
+			<option value="large"<?php if ($gallery_plus_size == 'large') echo ' selected="selected"'; ?>>Large</option>
 			<option value="full"<?php if ($gallery_plus_size == 'full') echo ' selected="selected"'; ?>>Full Size</option>
 			</select>
 		</td></tr>
@@ -96,9 +99,12 @@
 		Title: image title for both display title and caption.</label><br />
 		<label for="tcr3"><input id="tcr3" type="radio" name="gallery_plus_titlecaption" value="caption"<?php if ($gallery_plus_titlecaption == 'caption') echo ' checked="checked"'; ?> />
 		Caption: image caption for both display title and caption.</label><br />
+		<label for="tcr5"><input id="tcr5" type="radio" name="gallery_plus_titlecaption" value="download"<?php if ($gallery_plus_titlecaption == 'download') echo ' checked="checked"'; ?> />
+		Download: image title + download link for title and an empty caption.</label><br />
 		<label for="tcr4"><input id="tcr4" type="radio" name="gallery_plus_titlecaption" value="none"<?php if ($gallery_plus_titlecaption == 'none') echo ' checked="checked"'; ?> />
 		None: hide both display title and caption.</label>
 
+
 		</td></tr>
 
 		<tr valign="baseline">
@@ -123,6 +129,16 @@
 			Specify which image overlay package to use, such as lightbox.<br />
 			<strong>THIS PLUGIN DOES NOT INSTALL ANY OVERLAY PACKAGES.</strong>
 		</td></tr>
+		<tr valign="baseline">
+        <th scope="row"><?php _e('Gallery link size:') ?></th> 
+        <td><select name="gallery_plus_fullsize">
+			<option value="thumbnail"<?php if ($gallery_plus_fullsize == 'thumbnail') echo ' selected="selected"'; ?>>Thumbnail</option>
+			<option value="medium"<?php if ($gallery_plus_fullsize == 'medium') echo ' selected="selected"'; ?>>Medium</option>
+			<option value="large"<?php if ($gallery_plus_fullsize == 'large') echo ' selected="selected"'; ?>>Large</option>
+			<option value="full"<?php if ($gallery_plus_fullsize == 'full') echo ' selected="selected"'; ?>>Full Size</option>
+			</select>If linked to an image, what size should it be?
+		</td></tr>
+
 
 		<tr valign="baseline">
         <th scope="row"><?php _e('Only show gallery when single post:') ?></th>
